### DISCUSSION TITLE
Tag prefix/suffix with persisting versioning

### DISCUSSION THREAD

**Discussion by doteric** [semantic-release/semantic-release#2585] [Configuration] [answered]
Hello,

I would like to ask if it's possible to append a custom value to the release tag, but without affecting the versioning history.  
So for example let's say we have our image that has version 2.0.0, but the main thing it uses is some dependency and the version we use is 1.1.1. I would like to highlight that by adding that to the version so for example `2.0.0-1.1.1`. The next tag I want to be either `2.1.0-1.1.1` for example when the dependency version does not change, but I update something the main image itself. So the suffix only changes when it's changed manually (for example modifying some value in some configuration).  
Is this possible with the current sematnic-release or not?

I know there is `tagFormat`, but each change of it resets the version back to `1.0.0` so changing the version of `1.1.1` to `1.2.0` for example would result in `1.0.0-1.2.0` instead of the preferred `2.1.0-1.2.0`.

I also tried the other way around so using the dependency version as the base version so that we can start with `1.0.0` each time the dependency version is updated which could potentially also be fine, but then I stumbled upon a different problem. In this case all of the previous commits are taken into consideration not just the one from the previous release and the changelog is basically the whole history which is not what I'd like.  

Any recommendations on this? :D

**Comment by travi** [comment, marked as answer]
semantic-release is intended to simplify communication to consumers of your library about what the semantic impact of the changes in a new version would be if they upgraded to the new version. the versioning scheme that you are asking about is not semantic for _your library_, and therefore not supported by semantic-release. please keep in mind that semantic-release is not a generic workflow tool, but for helping to make semantic-versioning simpler.

**Reply by doteric** [reply]
Thanks @travi for the reply,
So I assume that this means that there is no such feature nor do you plan to have such introduced, correct? 😄

**Reply by travi** [reply]
correct. our supported workflows can be found at https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#branch-types, but what you describe does not fit those in a supported way. we do not have plans to support additional workflows at this time, especially any workflows that do not support semantic-versioning of your own library independently from dependencies.

**Reply by doteric** [reply]
@travi 
Getting back to this, would you be up for such feature if I created a PR? I don't think this is achievable via a plugin, because getting the last release cannot be modified with a plugin.

**Reply by travi** [reply]
sorry, what you are attempting to achieve is not in line with the goals of the project, so it is considered unsupported. again, semantic-release is not a generic workflow tool, so we are not interested in adding complexity to the project that would only exist to enable unsupported workflows.

**Reply by doteric** [reply]
Thank you @travi for the reply 🙇 
I fully understand you, but this would be a pretty simple addition. What would be needed is fetching the for tags having in mind the prefix and/or suffix and then later just adding one or both to the end result, nothing else. `${prefix}${semanticReleaseTag}${suffix}`, the only place it would need to be implemented is getting the latest tag and during tag creation and would be an optional thing.  The other option would be to add the ability for plugins to interact with the `getLatestRelease` step, what do you think?
